tribe_client/utils.py

```diff
@@ -407,17 +429,36 @@ def download_organism_public_genesets(organism, creator_username=None):
     downloaded from Tribe to get only the ones created by the user with
     this username.
 
+    request_params -- Optional argument, a dictionary of parameters to be
+    sent with the request to get gene sets from Tribe
+    (e.g. 'full_annotations').
+
     Returns:
     all_public_genesets -- A dictionary of GO, KEGG and DO terms available
     in Tribe for the specified organism (and optionally, creator_username).
     """
+    request_params['organism__scientific_name'] = organism
 
-    # *Note: Tribe does not like requests for more than 1500
-    # genesets at a time, so use this as the 'limit' parameter.
-    request_params = {
-        'show_tip': 'true', 'limit': '1500',
-        'organism__scientific_name': organism
-    }
+    if 'show_tip' not in request_params or not request_params['show_tip']:
+        request_params['show_tip'] = 'true'
+
+    # *Note: Tribe does not like requests for more than 1000 genesets
+    # at a time, so use this as the 'limit' parameter. This is not a hard
+    # limit, though - it varies based on the amount of data returned in
+    # the genesets.
+    try:
+        req_limit = int(request_params['limit'])
+        if req_limit <= 0:
+            request_params['limit'] = '1000'
+        elif req_limit > 1000:
+            logger.warning("We recommend setting the 'limit' parameter in "
+                           "the 'request_params' dictionary to less than "
```

Comment:
I think `less than` here should be `at most`.

Reply:
Ah good call, it is a subtle but important difference! I will change it.